File: 1b.py
#!/usr/bin/python3
import sys

tally = 0
with open(sys.argv[1], 'r') as fh_in:
    window = [int(fh_in.readline()) for _ in range(1, 4)]
    sumw = sum(window)
    for line in map(int, fh_in):
        newsum = line + sumw - window.pop(0)
        if newsum > sumw:
            tally += 1
        sumw = newsum
        window.append(line)
print(tally)

File: 1b.c
#include <stdio.h>

int main (int argc, char* argv[]) {
    int tally = -1;
    FILE* f;
    f = fopen(argv[1], "r");
    int window[3];
    int sum;
    for (int i = 0; i < 3; ++i) {
        fscanf(f, "%d", &window[i]);
        sum += window[i];
    }
    while (! feof(f)) {
        int line;
        fscanf(f, "%d", &line);
        int newsum = line + sum - window[0];
        if (newsum > sum) ++tally;
        window[0] = window[1];
        window[1] = window[2];
        window[2] = line;
        sum = newsum;
    }
    printf("%d\n", tally);
    return 0;
}
